### DISCUSSION TITLE
How to debug Virtual Machine issues?

### DISCUSSION THREAD

**Discussion by rwoolf** [elementary/triage#502] [Q&A]
I would like to know where to begin to help debug and fix issues with running Elementary OS in virtual machines.  I've used Elementary OS for years in virtual machines, but since EOS 7, drag/drop and copy paste no longer work between host and guest.  I am also experiencing other sporadic problems such as the OS hanging.  I've been a developer for many years, primarily coding in Java.  So I don't have experience yet in this type of development, or know where to start, but I can learn. I just don't know where to start to be able to debug such issues.  

Is there anyone that can point me in the right direction to be able to setup a debug environment to resolve issues with Elementary OS running on both VirtualBox and VMWare Workstation?  Any help and guidance appreciated.

**Comment by Africa-Hosting** [comment]
I have had the same problem like you, especially with the Oracle - Virtual Machine.
If you want to overcome the (copy/paste) (drag and drop) problem, just simply switch to vmWare Workstation. 
(as long as you still can) !!!
Last time i checked the Official vmWare Workstation Pro 17.5.2, was still available for download/sale.

I am not so sure now after BroadCom took over if they still sell it.

![image](https://github.com/elementary/triage/assets/172709457/2a391f77-525f-4b3e-a4a5-f69308875500)

as per above, with that Version it works absolutely like a dream.

Copy/Paste / Drag/Drop / Virtual Shared Drive.
Everything !

**Comment by Africa-Hosting** [comment]
By the way.
i remember the repository with sudo apt-get install virtualmachine (only relates to the older version)
rather follow this path

first, download the virtual machine from their website, make sure you download the extension pack too from https://www.virtualbox.org/
here the direct links to both :
https://www.virtualbox.org/wiki/Linux_Downloads
https://download.virtualbox.org/virtualbox/7.0.18/Oracle_VM_VirtualBox_Extension_Pack-7.0.18.vbox-extpack

then execute  :

`sudo gdebi virtualbox-7.0_7.0.18-162988~Ubuntu~jammy_amd64.deb && sudo apt install libelf-dev`
`sudo usermod -aG vboxusers administrator`
(replace administrator with the user of the machine (your user name you chose when you installed OS 7

`sudo apt install build-essential linux-kernel-headers`
`sudo apt install --reinstall gcc-12`

execute the application
once you started the application, load the extension pack that you just recently downloaded

execute the machine of your choice / windows etc.
make sure once you are in windows or any other os, that you load the guest drivers

see the picture below for your reference
![image](https://github.com/elementary/triage/assets/172709457/386dc479-0861-49e5-bfb4-acbce0b57d0a)

like i said. this will ONLY partially work. rather use vmware workstation.

good luck
